src/main/java/org/cyclonedx/generators/xml/AbstractBomXmlGenerator.java

```diff
@@ -184,6 +190,72 @@ private void createComponentNode(Node parent, Component component) {
         processExtensions(componentNode, component);
     }
 
+    void createVulnerabilitiesNode(Node parent, List<Vulnerability1_0> vulnerabilities) {
+        if (vulnerabilities != null && !vulnerabilities.isEmpty()) {
+            final Element vulnsNode = createElement(parent, "v:vulnerabilities");
+            for (Vulnerability1_0 vuln : vulnerabilities) {
+                createVulnerabilityNode(vulnsNode, vuln);
+            }
+        }
+    }
+
+    private void createVulnerabilityNode(Node parent, Vulnerability1_0 vulnerability) {
+        final Element vulnNode = createElement(
+            parent, "v:vulnerability", null, new Attribute("ref", vulnerability.getRef())
+        );
+
+        if (vulnerability.getId() != "") {
+            final Element id = createElement(vulnNode, "v:id");
+            id.setTextContent(vulnerability.getId());
+        }
+
+        if (vulnerability.getDescription() != "") {
+            final Element description = createElement(vulnNode, "v:description");
+            description.setTextContent(vulnerability.getDescription());
+        }
+
```

Comment:
similar to above, what about `source` ? https://github.com/CycloneDX/specification/blob/master/schema/ext/vulnerability-1.0.xsd#L179

Reply:
outdated now